conduit-combinators/Data/Conduit/Combinators.hs

```diff
@@ -1630,6 +1632,39 @@ slidingWindowC sz = go (max 1 sz) mempty
                        Just x -> go (n-1) (Seq.snoc st x)
 STREAMING(slidingWindow, slidingWindowC, slidingWindowS, sz)
 
+
+-- | Split input into chunk of size 'chunkSize'
+--
+-- The last element may be smaller than the 'chunkSize' (see also
+-- 'chunksOfExactlyE' which will not yield this last element)
+--
+-- @since 1.1.2
+chunksOfE :: (Monad m, Seq.IsSequence seq) => Seq.Index seq -> Conduit seq m seq
+chunksOfE chunkSize = chunksOfExactlyE chunkSize >> (await >>= maybe (return ()) yield)
+
+-- | Split input into chunk of size 'chunkSize'
+--
+-- If the input does not split into chunks exactly, the remainder will be
+-- leftover (see also 'chunksOfE')
+--
+-- @since 1.1.2
+chunksOfExactlyE :: (Monad m, Seq.IsSequence seq) => Seq.Index seq -> Conduit seq m seq
+chunksOfExactlyE chunkSize = await >>= maybe (return ()) start
+    where
+        start b
+            | onull b = chunksOfE chunkSize
+            | Seq.lengthIndex b < chunkSize = continue (Seq.lengthIndex b) [b]
```

Comment:
Arguably instead of `[b]` and then reversing the list, you could use a difference list approach with `b:`. It's not absolutely necessary though.